dashboard/app/models/backpack.rb

```diff
@@ -13,4 +13,23 @@
 #  index_backpacks_on_user_id  (user_id)
 #
 class Backpack < ApplicationRecord
+  belongs_to :user
+  validates_uniqueness_of :storage_app_id
+  validates_uniqueness_of :user_id
+
+  def self.find_or_create(user_id, ip)
+    backpack = find_by_user_id(user_id)
+    unless backpack
+      # Create a storage app for this user's backpack
+      storage_app = StorageApps.new(get_storage_id)
```

Comment:
Will the record created in `StorageApps` have `user_id` = nil? If so, is that intentional?

Reply:
storage apps doesn't have a user id field, it instead has `storage_id` (which is unique per user), which is what we are passing in with `get_storage_id`

Reply:
Sorry meant the record in `user_storage_ids`, which I think will have a user ID in the actual controller action/test (b/c we have a current_user), and will not have a user ID for the model test you added, or if we ever created backpacks outside the context of a controller (b/c no current_user)? Probably doesn't matter

https://github.com/code-dot-org/code-dot-org/blob/65e1499521235ef7a35053b54a93b903ad108729/shared/middleware/helpers/storage_id.rb#L95

https://github.com/code-dot-org/code-dot-org/blob/65e1499521235ef7a35053b54a93b903ad108729/shared/middleware/helpers/storage_id.rb#L5

Reply:
ah gotcha. I'll update this to explicitly get the storage_id for the given user_id